src/main/java/fshop/service/FoodService.java

```diff
@@ -0,0 +1,34 @@
+package fshop.service;
+
+import fshop.service.strategy.StrategyImpl;
+import fshop.service.strategy.StrategyPurchase;
+import fshop.service.strategy.StrategyReturn;
+import fshop.service.strategy.StrategySupply;
+
+public class FoodService {
+    private static final String COMA = ",";
+    private static final String PURCHASE = "p";
+    private static final String RETURN = "r";
+    private static final String SUPPLY = "s";
+
+    public Integer selectStrategy(String operations, int valueFromDb) {
+        StrategyImpl strategy = new StrategyImpl();
+        if (operations.substring(0, operations.indexOf(COMA)).equals(PURCHASE)) {
+            strategy.setStrategy(new StrategyPurchase());
+            return strategy.execute(valueFromDb, getValueFromFile(operations));
+        }
+        if (operations.substring(0, operations.indexOf(COMA)).equals(RETURN)) {
+            strategy.setStrategy(new StrategyReturn());
```

Comment:
usage of strategy is not correct. The code should be a-la:
Strategy strategy = new StrategyImpl(Map<String, OperationHandler>);
startegy.get(operation).do(....);
It is help to avoid usage of if-loops.

Reply:
Would Only add that you Should create this map in your main method and you Can pass StrategyImpl here via constructor